src/object/obj_tx.c

```diff
@@ -1208,6 +1204,16 @@ dc_tx_classify_common(struct dc_tx *tx, struct daos_cpd_sub_req *dcsr,
 		if (rc != 0)
 			goto out;
 
+		if (reasb_req != NULL && reasb_req->tgt_bitmap != NIL_BITMAP &&
+		    isclr(reasb_req->tgt_bitmap, shard->do_shard - start)) {
+			D_DEBUG(DB_TRACE, "idx %d shard %u start %d skip\n", idx,
+				shard->do_shard, start);
+			obj_shard_close(shard);
+			continue;
+		}
```

Comment:
just confirm any necessary to move this section down compared to original place? thx
seems original code that if the shard is not in bitmap then need not shard open

Reply:
hmm, sorry, I can not think of the reason, probably for debugging purpose or making the EC object is still valid before any modification.

Anyway let's revert this change, and I will see if I can figure out why I made this change later. Thanks.